### DISCUSSION TITLE
kj::Own::get const propagation

### DISCUSSION THREAD

**Discussion by ericjacobs** [capnproto/capnproto#2098] [Ideas]
In kj/memory.h, we have

```
  inline T* get() { return ptr; }
  inline const T* get() const { return ptr; }
```
which says that a const `Own<T>` can only be used to access the `T` as const. This is an annoyance compared with `std::unique_ptr` and results in unnecessary const/mutable propagation. Maybe this could be relaxed before the API is finalized?

**Comment by kentonv** [comment]
Transitive constness is intentional. This allows const qualifiers to be used to enforce thread safety, which is much more valuable than traditional const qualifier use.

See: https://github.com/capnproto/capnproto/blob/v2/style-guide.md#constness

> Maybe this could be relaxed before the API is finalized?

Not sure what you mean, this API has been stable for 10+ years now.
